shared/actions/wallets.tsx

```diff
@@ -1112,44 +1112,105 @@ const gregorPushState = (_, action: GregorGen.PushStatePayload) =>
     show: !action.payload.state.find(i => i.item.category === Constants.airdropBannerKey),
   })
 
-const rpcAssetToAssetDescription = (asset: RPCStellarTypes.Asset): Types.AssetDescription =>
-  Constants.makeAssetDescription({
-    code: asset.code,
-    issuerAccountID: asset.issuer,
-    issuerName: asset.issuerName,
-    issuerVerifiedDomain: asset.verifiedDomain,
-  })
+const assetDescriptionOrNativeToRpcAsset = (
+  asset: 'native' | Types.AssetDescription
+): RPCStellarTypes.Asset =>
+  asset === 'native'
+    ? {
+        code: '',
+        desc: '',
+        infoUrl: '',
+        infoUrlText: '',
+        issuer: '',
+        issuerName: '',
+        type: 'native',
+        verifiedDomain: '',
+      }
+    : {
```

Comment:
:sheep: might be nice to have this be one block, since most of the fields are the same in either case